### ISSUE TITLE
Safelist pattern warning on own classes.

### ISSUE THREAD

**Issue by andershagbard** [tailwindlabs/tailwindcss#8511]
**What version of Tailwind CSS are you using?**

v3.0.24

**What build tool (or framework if it abstracts the build tool) are you using?**

webpack 5.72.1

**What version of Node.js are you using?**

v16.14.0

**What operating system are you using?**

macOS

**Describe your issue**

If I whitelist some of my "custom" classes using a pattern, Tailwind will warn me every time i build.

```
warn - The safelist pattern `/my-class--(.)*/` doesn't match any Tailwind CSS classes.
warn - Fix this pattern or remove it from your `safelist` configuration.
warn - https://tailwindcss.com/docs/content-configuration#safelisting-classes
```

This warning can cause some confusion, and shouldn't be added if it matches one of your own classes.

**Comment by adamwathan** [comment]
Hey! Please include a reproduction when you open an issue, it's a requirement and like mentioned in the issue template.

That said I looked into this quickly and everything is working as expected.

Give this CSS input, Tailwind outputs no warning:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer utilities {
  .my-class--foo {
    color: black;
  }
}
```


Give this CSS input, Tailwind _does_ output a warning:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

.my-class--foo {
  color: black;
}
```

That's because the class isn't registered within Tailwind using a `@layer`, so it is always included no matter what, not just generated when needed. That means you don't actually need to safelist it at all.

This is all covered in the documentation here:

https://tailwindcss.com/docs/adding-custom-styles#removing-unused-custom-css

TL;DR — you don't need to safelist things unless you register them in a `@layer`.

**Comment by andershagbard** [comment]
Thought this was more trivial than it seems. My classes are within a @layer. I'll include a repo later.

**Comment by 1aurabrown** [comment]
@adamwathan Coming back to this issue—I'm having the same issue, when including custom classes in the "base" layer (within `@layer base { ... }`). Moving the styles to "utilities" solves the problem.

My use-case is that I am trying to style classes present on elements that are injected into the website (a Shopify project) by a third-party widget. The classes will not be found anywhere in my project files and need to be explicitly whitelisted. Is it inappropriate to declare these styles in the `base` layer? Why is the whitelisting logic different for different layers?

Thanks!
